app/presenters/tree_builder_report_reports_class.rb

```diff
@@ -5,6 +5,7 @@ class TreeBuilderReportReportsClass < TreeBuilder
 
   def x_get_tree_r_kids(object, count_only)
     objects = MiqReportResult.with_current_user_groups_and_report(object.id).to_a
```

Comment:
@ZitaNemeckova yes. we ussualy tends to have everything in sql queries as it is possible and in addition you don't have to create new array and walk  thru it.

and I see these two method are doing same thing, what about moving into one ?

thanks :)